src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -958,6 +972,23 @@ export class CodeWindow extends BaseWindow implements ICodeWindow {
 		}
 	}
 
+	private async startCollectingJScallStacks(): Promise<void> {
+		if (this.JScallStackCollector) {
+			return;
+		}
+		this.JScallStackCollector = setInterval(async () => {
```

Comment:
We have a `RunOnceScheduler` utility for this that is a bit easier to work with.